scripts/unity/header_prepare.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+import re
+import argparse
+
+
+def header_prepare(in_file, out_file, out_wrap_file):
+    with open(in_file) as f_in:
+        content = f_in.read()
+
+    # change static inline functions to normal function declaration
+    static_inline_pattern = re.compile(
+        r'((__deprecated )?)((static inline )|(static ALWAYS_INLINE )|'
```

Comment:
This regex is vulnerable to unintended additional spaces. as example if programmer by accident writes:
```
static  inline
``` 
with 2 spaces instead of one, it will fail.
And for the programmer the reason why `header_prepare.py` generates wrong (missing) output will be hard to track down, as a space too much in a header file might not be the first place to consider.

Second thing: Low priority.
regex is also vulnerable to line breaks in function declaration.
Although it's not used in NCS, I have seen that principle in code elsewhere, so it's up to you to decide if that should be handled as well.

Reply:
replaced spaces with `\s+`.